contracts/mocks/CompactMemProofMock.sol

```diff
@@ -0,0 +1,49 @@
+pragma experimental "v0.5.0";
+pragma experimental ABIEncoderV2;
+pragma solidity 0.4.24;
+
+import {EthereumRuntime} from "../EthereumRuntime.sol";
+import {EVMMemory} from "../EVMMemory.slb";
+import {MerkleProof} from "./MerkleProof.slb";
+
+contract CompactMemProofMock {
+    using EVMMemory for EVMMemory.Memory;
+
+    EthereumRuntime public ethRuntime;
+
+    constructor(address _ethRuntime) public {
+        ethRuntime = EthereumRuntime(_ethRuntime);
+    }
+
+    function verify(
+        bytes code,
+        bytes memory data,
+        uint[4] memory intInput,
+        uint[] memory stack,
+        bytes memory mem,
+        bytes32[] memProof,
+        bytes32 rootMemHash
+    ) public view returns(bool){
+        uint[] memory accounts;
+        bytes memory accountsCode;
+        bytes32 logHash;
+        EthereumRuntime.Result memory res = ethRuntime.execute(
+            code, 
+            data, 
+            intInput, 
+            stack, 
+            mem, 
```

Comment:
I am confused about passing `mem` here.
I expected that we would only pass the `memProof`, feed it into the runtime, and if the runtime would modify the leaf, the proof/root would update. if the runtime would try to access any other element than the leaf in the memory, then the execution would throw/revert. 
The advantage would be that even if the memory is huge, we would only need to pass logarithmic size of data to the execution.
will try to prepare a proposal how to adjust the code.

Reply:
`mem` is the leaf value here. 

For example, say if you had a total memory size of 20 words but the operation you needed to verify only requires the 3rd word from the begining. 
In this case you would only pass in the first 4 words in the `mem`. The operation would be executed and a merkle Hash of the resulting memory calculated. This hash can then be verified using the provided `memProof`. If the resulting roothash equals `rootMemHash` then the party(solver or challenger)  has submitted the correct result.

Reply:
if the code would require to read or write the 18 words, how big would `mem` need to be?

Reply:
do you see a way to make this static sized `log(sizeIn32byteWords)` no matter which position is accessed? MSTORE and MLOAD opcodes only operate on a single 32 byte word. let's say we want to have memory size of multiple megabytes, then accessing the last element would exceed the gas limit to create a proof.

In this drawing i try to show that if you split the memory array into 32 byte slots, and merkelize them, writing to a proven leave anywhere in the tree only requires `log(sizeIn32byteWords)` elements.
https://docs.google.com/drawings/d/15NOju-EwJpn3JLX4zeAFvifIBGOY5D21BBwOc6cRRr0

Reply:
@vyomshm I the part above ^, involves more changes(a sparse memory setup), but I covered that in #58 .
The bounty scope was/is a bit chaotic, as the scope wasn't not clearly defined enough.
I'm going to implement your PR in a later state with the sparse memory setup and consider this issue done. Unless you have objections? 😄

Reply:
I see. That's okay with me 😄

Reply:
@vyomshm submit on gitcoin :)